src/main/java/org/alfresco/bm/rest/UpdateNodeProperties.java

```diff
@@ -0,0 +1,128 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.bm.rest;
+
+import javax.json.Json;
+import javax.json.JsonObject;
+import javax.json.JsonObjectBuilder;
+
+import org.alfresco.bm.common.EventResult;
+import org.alfresco.bm.driver.event.Event;
+import org.alfresco.bm.user.UserData;
+import org.alfresco.bm.user.UserDataService;
+import org.alfresco.rest.core.RestWrapper;
+import org.alfresco.rest.model.RestNodeModel;
+import org.alfresco.utility.model.FileModel;
+import org.alfresco.utility.model.UserModel;
+
+import com.mongodb.BasicDBObjectBuilder;
+import com.mongodb.DBObject;
+
+/**
+ * <b>INPUT:   </b>A site id and a random site member of the site, along with a nodeRef of a file created in a previous event by the same user.<br/>
+ * <b>ACTIONS: </b>Updates some properties of the node with the corresponding nodeRef.<br/>
+ * <b>OUTPUT:  </b>Propagates the same site id and member to the next event, along with the noderef of the file.
+ * 
+ * @author Ancuta Morarasu
+ */
+public class UpdateNodeProperties extends RestTest 
+{
+    public static final String EVENT_PROPERTIES_UPDATED = "eventNodePropertiesUpdated";    
+    
+    private UserDataService userDataService;
+    private String eventNodePropertiesUpdated;
+    
+    public UpdateNodeProperties() 
+    {
+        this.eventNodePropertiesUpdated = EVENT_PROPERTIES_UPDATED;
+    }
+    
+    public void setEventNodePropertiesUpdated(String event)
+    {
+        this.eventNodePropertiesUpdated = event;
+    }
+
+    public void setUserDataService(UserDataService userDataService) 
+    {
+        this.userDataService = userDataService;
+    }
+
+    @Override
+    protected EventResult processEvent(Event event) throws Exception 
+    {
+        super.suspendTimer();
+        
+        DBObject dataObj = (DBObject) event.getData();
+        if (dataObj == null)
+        {
+            throw new IllegalStateException("This processor requires data with field " + SiteContextConstants.SITE_ID);
+        }
+        String site = (String) dataObj.get(SiteContextConstants.SITE_ID);
+        String member = (String) dataObj.get(SiteContextConstants.SITE_MEMBER);
+        String nodeRef = (String) dataObj.get(SiteContextConstants.NODE_ID);
+        
+        //Get the user for the selected site member
+        UserData userdata = userDataService.findUserByUsername(member);
+        if (userdata == null)
+        {
+            return new EventResult("Failure: The picked user doesn't exist anymore: " + member, false);
+        }
+        
+        UserModel userModel = new UserModel();
+        userModel.setUsername(userdata.getUsername());
+        userModel.setPassword(userdata.getPassword());
+        
+        FileModel file = new FileModel();
+        file.setNodeRef(nodeRef);
+        
+        JsonObjectBuilder properties = Json.createObjectBuilder()
```

Comment:
This text could be generated using the text generator: https://github.com/Alfresco/alfresco-text-gen
That way, the search index is nicely distributed for title and description.

Reply:
I used the `TextGenerator` to create the title and description. But since in this particular scenario the node that is modified is also deleted at the end, I don't think will influence the search. Still a good reference and will keep it in mind for the search scenario.